client.go

```diff
@@ -115,16 +142,50 @@ func (c *Client) handleResponse(apiResponse *http.Response, response interface{}
 		var errorResponse ErrorResponse
 		err := json.Unmarshal(body, &errorResponse)
 		if err != nil {
-			return fmt.Errorf("api error(%v): no body", apiResponse.StatusCode)
+			return NewMapboxError(apiResponse.StatusCode, "")
 		}
 
-		return fmt.Errorf("api error(%v): %v", apiResponse.StatusCode, errorResponse.Message)
+		// If rate limited, hold off till the next X-Rate-Limit-Reset
+		if apiResponse.StatusCode == 429 {
+			resetUnix, err := strconv.Atoi(apiResponse.Header.Get("X-Rate-Limit-Reset"))
+			if err == nil {
+				// c.rateLimitMutex.Lock()
+				// defer c.rateLimitMutex.Unlock()
```

Comment:
```suggestion
				c.rateLimitMutex.Lock()
				defer c.rateLimitMutex.Unlock()
```

how come this is commented?

Reply:
That should not be commented 🤦‍♂️ That was me seeing race conditions with and without this, sorry about that.